apis/python/src/tiledbsoma/_query.py

```diff
@@ -407,9 +409,11 @@ def obs_scene_ids(self) -> pa.Array:
         try:
             obs_scene = self.experiment.obs_spatial_presence
         except KeyError as ke:
-            raise KeyError("Missing obs_scene") from ke
+            raise KeyError(
+                "No obs_spatial_presence dataframe in this experiment."
+            ) from ke
         if not isinstance(obs_scene, DataFrame):
-            raise TypeError("obs_scene must be a dataframe.")
+            raise TypeError("obs_spatial_presence must be a dataframe.")
```

Comment:
```suggestion
            raise TypeError("obs_spatial_presence must be a dataframe; got {type(obs_scene)}.")
```
and likewise below (same pattern)